assets/src/edit-story/karma/copyAndPaste.cuj.karma.js

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import useInsertElement from '../components/canvas/useInsertElement';
+import { addBackgroundImage } from '../components/dropTargets/karma/background.karma';
+import { useStory } from '../app/story';
+import { Fixture } from './fixture';
+
+/**
+ * Takes an HTMLCollection and sequentially performs an
+ * async action on each element in the collection. 
+ 
+ * Call order of operations is based on the order of
+ * the elements in the collection
+ *
+ * @param {HTMLCollection} htmlCollection - collection of html elements to perform operation on.
+ * @param {(el: HTMLElement, i: number) => Promise<void>} op - async operation
+ * @return {Promise<void>}
+ */
+function sequencedForEach(htmlCollection, op) {
+  return new Promise((res) => {
+    Array.from(htmlCollection)
+      .map((htmlElement, i) => async () => {
+        await op(htmlElement, i);
+      })
+      .reduce((p, fn) => p.then(fn), Promise.resolve())
```

Comment:
This right here is a crash course in promises and async. But super cool.

Reply:
oh yea. took me a sec to figure out. might be nice to put a little comment of the transformation on there:
```
// [p1, p2, p3, p4] -> promise.resolve().then( p1).then( p2).then( p3).then( p4)
```